### PR TITLE
render messages that summarize K8 container state

### PR DESCRIPTION
**Pull request by claymccoy**
We now get the images from the "After".
By parsing the k8 JSON we could add some more info about  the state (why is it waiting, when did it start...). But this seemed like a good first step.

I couldn't use After.images yet. It would not populate in the types.ts on compile, I think because it does not exist in prod yet. The way that it is written, an image array will be a trivial change.
I couldn't use the automation-client from master due to compile errors, I had to use the one from the original PR surprisingly. I don't know what automation-client should be used, I expected the automation-client used in master to contain everything that the one from my PR last week had.
Probably due to the automation-client dep, I had to change the call to initMemoryMonitoring in atomist.config.ts in order to compile. This change has nothing to do with the real code changes, and I don't expect it to stick.
I think I need to change the other similar graphql subscriptions to have after.images too, even though they are unused there.

### CODE REVIEW

**Comment by ddgenome** [comment]
I'm going to try to fix this forward. It may involve rebasing.

File: src/atomist.config.ts [hunk 1]
```diff
@@ -290,4 +290,4 @@ export const configuration: any = {
 };
 
 // For now, we enable a couple of interesting memory and heap commands on this automation-client
-initMemoryMonitoring(`${appRoot.path}/heap`, false);
+initMemoryMonitoring(`${appRoot.path}/heap`);
```
Comment (ddgenome) [inline src/atomist.config.ts]: Is this change still necessary?

File: src/handlers/event/push/K8PodToPushLifecycle.ts [hunk 2]
```diff
@@ -18,15 +18,15 @@ import { PushLifecycleHandler } from "./PushLifecycle";
 export class K8PodToPushLifecycle extends PushLifecycleHandler<graphql.K8PodToPushLifecycle.Subscription> {
 
     protected extractNodes(event: EventFired<graphql.K8PodToPushLifecycle.Subscription>):
-        graphql.PushToPushLifecycle.Push[] {
+        graphql.K8PodToPushLifecycle.Pushes[] {
```
Comment (ddgenome) [inline src/handlers/event/push/K8PodToPushLifecycle.ts]: Most of the other `extractNodes` methods return `graphql.PushToPushLifecycle.Push[]`, but I'm not sure why. This seems more correct to me.

File: src/handlers/event/push/rendering/PushNodeRenderers.ts [hunk 3]
```diff
@@ -491,61 +491,72 @@ export class ApplicationNodeRenderer extends AbstractIdentifiableContribution
...
-            .forEach(t => t.containers.forEach(con => {
-                const image = con.image;
-                if (images[image]) {
-                    images[image].push(con);
-                } else {
-                    images[image] = [con];
-                }
-            })));
-
+        const images = [push.after.image];
```
Comment (ddgenome) [inline src/handlers/event/push/rendering/PushNodeRenderers.ts]: Most of the other `render` methods use `context.lifecycle.extract` to get their main property. Why is this one different?

Comment (ddgenome) [reply src/handlers/event/push/rendering/PushNodeRenderers.ts]: Also, the new commit-[]-images is in the schema now, so the subscription and this should probably be changed to `images`.

File: src/atomist.config.ts (hunk 1 above)
Comment (claymccoy) [reply src/atomist.config.ts]: no

File: src/handlers/event/push/K8PodToPushLifecycle.ts (hunk 2 above)
Comment (claymccoy) [reply src/handlers/event/push/K8PodToPushLifecycle.ts]: I don't know either.
This change was actually made by cd in the original PR.

File: src/handlers/event/push/rendering/PushNodeRenderers.ts (hunk 3 above)
Comment (claymccoy) [reply src/handlers/event/push/rendering/PushNodeRenderers.ts]: I have the same question. I don't know why this one didn't fit the mold. But it was different before I started making changes to it. I'd love it if @cdupuis clarified some of this here.
